src/client.ts

```diff
@@ -281,26 +281,28 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     return this.isUsingSdkKey;
   }
 
-  public onReady(config: { timeout?: number } = {}): Promise<OnReadyResult> {
+  public onReady(
+    config: { timeout?: number; segmentOptions?: optimizely.OptimizelySegmentOption[] } = {}
+  ): Promise<OnReadyResult> {
     let timeoutId: number | undefined;
     let timeout: number = DEFAULT_ON_READY_TIMEOUT;
     if (config && config.timeout !== undefined) {
       timeout = config.timeout;
     }
-
     const timeoutPromise = new Promise<OnReadyResult>(resolve => {
       timeoutId = setTimeout(() => {
         resolve({
           success: false,
           reason: 'TIMEOUT',
           message:
-            'failed to initialize onReady before timeout, either the datafile or user info was not set before the timeout',
+            'failed to initialize onReady before timeout, either the datafile, user info or qualified segments was not set before the timeout',
           dataReadyPromise: this.dataReadyPromise,
         });
       }, timeout) as any;
     });
 
-    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(res => {
+    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(async res => {
+      await this.fetchQualifiedSegments(config?.segmentOptions);
```

Comment:
`fetchQualifiedSegment` should be called only on datafile success not on timeout. Also can you check if this the caller is blocked until `fetchQualifiedSegments` completed. It looks like async call.

Reply:
Thanks. added a check timeout check on this. And yes this the caller is blocked until fetchQualifiedSegments completed